### ISSUE TITLE
isPlayerCrosshairVisible returns true at the start of aiming

### ISSUE THREAD

**Issue by ghost** [multitheftauto/mtasa-blue#3760]
### Describe the bug

Hello.
I'm glad that this new function added to the game, but in this state its the equivalent of getControlState("aim_weapon")
The only difference is that when you run, so your charachter won't be able to aim, it won't return as true, unlike the controlState. 

I would like to make a script which would prevent players from shooting when their crosshair is not fully visible.
Without the crosshair, the players can fast aim + fast shoot to acquire no-recoil on weapons. Also these shots are shooted at head level so if a server uses instant death on headshots they can easily kill players with this.
This is a problem especially with sawed-off shotgun. On most server this weapon is the best close combat weapon, on my server 90% of the fights are made with sawed-off. The players somehow learnt a way on how to properly run in circles + do this fast aim + shoot thing, so they don't really have to aim, they just need to time their charachter's rotation where their opponent is. I am unable to show this in progress, but I can show you how the no-recoil (clearly visible with UZI) and head-level shooting works. On most cases the second shoot will be the problem, but as you can see in the video I was able to do it with all 4 bullets.

Video: https://youtu.be/nzknvCRo06w?t=23
From 0:23 Its clearly visible that the second shoot goes to head level, and has base recoil only. With crosshair the recoil/spreading would be higher especially with poor skills.
At 0:37 and 0:39 I was able to shoot the first bullet at head level & with only base recoil as well.
When I switched to UZI, you can see it has no recoil at all. 

### Steps to reproduce

```lua
--- Put weapons here, set to true if you want hipfire blocked for it
-- @warning don't use 0 or 36 as they are already blocked
local antiHipfireWeapons = {
    [22] = true,
    [23] = true,
    [24] = true,
    [25] = true,
    [26] = true,
    [27] = true,
    [28] = true,
    [29] = true,
    [30] = true,
    [31] = true,
    [32] = true,
    [33] = true,
    [38] = true,
}

-- So we can't lose track of our state if some other script decides to use toggleControl
-- or when this script gets stopped we need to reset the state
-- this is set true when we're using it, false when we're not
local fireControlLock = false
-- Time since last antiHipfire check
local lastTick = 0
-- Time interval between each hipfire check
local checkInterval = 300

--- Disable hipfire for certain weapons
local function antiHipfire()
    local curTick = getTickCount()
    if curTick - lastTick > checkInterval then
        if getPedOccupiedVehicle(localPlayer) then
            return
        end

        local wp = getPedWeapon(localPlayer)

        -- Stop here if it's not anti hipfire weapon
        if not antiHipfireWeapons[wp] and fireControlLock then
            toggleControl("fire", true)
            fireControlLock = false
            return
        end

        --local aiming = getPedControlState(localPlayer, "aim_weapon")
        local aiming = isPlayerCrosshairVisible()
        if antiHipfireWeapons[wp] and isControlEnabled("fire") ~= aiming then
            toggleControl("fire", aiming)
            fireControlLock = not aiming
        end

        lastTick = curTick
    end
end
addEventHandler("onClientPreRender", root, antiHipfire)

-- If we stop this resource make sure we undo toggleControl
addEventHandler("onClientResourceStop", resourceRoot, function ()
    if fireControlLock then
        toggleControl("fire", true)
    end
end)
```

Use this code, to test it. Made by Dutchman. 

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not security related.

**Comment by Dutchman101** [comment]
Hey @derxgbb, i gotta post this here because i dont know of any other way to reach you.. *assuming* you are "Kerek" on discord, it appears you'd be the guy known as Asturel from GTO freeroam as well.

I emailed the old astur4l address, but i dont know if you still use it.

Please take notice of what i wanted to say: [https://i.imgur.com/QubX7le.png](https://i.imgur.com/QubX7le.png)
I don't want a server like GTO to be shut down over a misunderstanding..

After you read this, we can clear this.

**Comment by derxgbb** [comment]
@Dutchman101 I wrote a book of long answer to you, but I don't want to place it in here, and on DC I can't message you. If you add me back, I can send it. 

But anyway I am not asturel, but you are not far away of thinking.

**Comment by FileEX** [comment]
I'll look into it. This function is based on the same principle as the original GTA code, i.e. on checking the camera mode (the characteristic zoom when aiming). Can you provide exact steps so I can reproduce your problem? Should I enable the "fastfire" glitch?

**Comment by ghost** [comment]
@FileEX Very sorry, that I didn't provide enough information.

I enabled only 'quickreload' in order to mimic the environment of our main server.

```lua
setGlitchEnabled('quickreload', true)
```

This code just writes the current state of the three functions:

```lua
addEventHandler("onClientPreRender", root, function() 
    dxDrawText("isPlayerCrosshairVisible : " .. tostring(isPlayerCrosshairVisible()), 35, 250, 300, 300, 0xFFFFFFFF,1.4)
    dxDrawText("getControlState(\"aim_weapon\") : " .. tostring(getControlState("aim_weapon")), 35, 270, 300, 300, 0xFFFFFFFF,1.4)
    dxDrawText("isPedDoingTask(\"TASK_SIMPLE_USE_GUN\") : " .. tostring(isPedDoingTask(localPlayer, "TASK_SIMPLE_USE_GUN")), 35, 290, 300, 300, 0xFFFFFFFF,1.4)
end)
```

This sawed shotgun is very popular among players, I saw it in other servers as well. I should have mentioned, that we use quickreload, in order to have so called 'skilled' fights between people. 
Maybe there is no way to detect when the crosshair is fully visible. If this is how it works in the singleplayer, then I guess we have to implement a custom crosshair. Maybe if there is a way of detecting when the crosshair is drawn instead of camera modes, but I don't know the game code, so I won't make bad speculations.

**Comment by FileEX** [comment]
Fixed and tested in #3762
